src/main/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImpl.java

```diff
@@ -47,6 +60,15 @@ public AcspProfileData create(AcspProfileSpec spec) throws DataException {
             }
             profile.setAmlDetails(amlDetailsList);
         }
+        if ("sole-trader".equals(spec.getType())) {
```

Comment:
You might consider using 
`Objects.equals("sole-trader", spec.getType())`
here to add some type safety around potential null values. While it's fine as is in a test, it's a good habit to develop for general code safety